### PR TITLE
Unit spec

### PR DESCRIPTION
**Pull request by naveed-ahmad**
- Associations
- DB columns  and indexes
- Setup rubocop and related gems

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -1,89 +1,99 @@
...
 git_source(:github) do |repo_name|
   repo_name = "#{repo_name}/#{repo_name}" unless repo_name.include?("/")
   "https://github.com/#{repo_name}.git"
 end
 
 # Bundle edge Rails instead: gem 'rails', github: 'rails/rails'
-#gem 'rails', '4.1.1'
-gem 'rails', '~> 5.0.1'
+# gem 'rails', '4.1.1'
+gem "rails", "~> 5.0.1"
```
[the remark this answers is not in the document]
Comment (mmahalwy) [reply Gemfile]: Coming from here: https://github.com/rails/rails/blob/master/.rubocop.yml#L123

I think we should enforce single quotes TBH.

File: app/controllers/v3/verses_controller.rb [hunk 2]
```diff
@@ -21,94 +23,94 @@ def show
...
-  def pagination_dict(object)
-    {
-      current_page: object.current_page,
-      next_page: object.next_page,
-      prev_page: object.prev_page,
-      total_pages: object.total_pages,
-      total_count: object.total_count
-    }
-  end
+    def pagination_dict(object)
```
[the remark this answers is not in the document]
Comment (naveed-ahmad) [reply app/controllers/v3/verses_controller.rb]: @mmahalwy this is from rails https://github.com/rails/rails/blob/master/.rubocop.yml#L63 but agreed no need to indent methods after modifier. Will fix that.

**Comment by naveed-ahmad** [comment]
rebuild

**Comment by naveed-ahmad** [comment]
rebuild

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:32802

**Pull request merged**